docs/enhancements/4-nodes-cluster-deployment.md

```diff
@@ -0,0 +1,159 @@
+---
+title: 4-nodes-cluster-creation
+authors:
+  - "@flaper87"
+creation-date: 2022-04-14
+last-updated: yyyy-mm-dd
+---
+
+# Allow for a 4-nodes cluster to be created
+
+# Summary
+
+**Note:** Throughout this enhancement, a 4-nodes cluster refers to a 3 schedulable masters and 1 worker cluster.
+
+There are cases where customers may just have 4 nodes available and no chance to add more. This was
+one of the cases in the ZTP for Factory Workflow where it was required to deploy a cluster in an
+enclosure that only had 4 nodes. The goal was to use all of the blades rather than just 3 of them.
+
+For now, this is achieved with a 2 step deployment, first 3 masters and then add a worker node. It
+would be more time efficient as well as easier and more reliable to be able to do this deployment in
+one step, day0.
+
+# Motivation
+
+Assisted service has built-in validations that prevent the deployment of a 4-nodes cluster. This is
+to prevent the deployment of topologies that are not part of the OCP standard test matrix. A 4-nodes
+cluster doesn't take schedulable masters into consideration when there are workers being deployed,
+which causes the installer to chose an `InfrastructureTopology` that is SingleReplica. This means
```

Comment:
This isn't accurate, for example with the current way we install 4 nodes cluster (3 control-plane nodes, add a worker) the ingress will get scheduled on the control-plane (@omertuc correct me if I'm wrong here)

Reply:
@omertuc do you know what will happen to the ingress deployment in case the masters are schedulable and the InfrastructureTopology is highly available but there is a single worker node?

Reply:
>@omertuc do you know what will happen to the ingress deployment in case the masters are schedulable and the InfrastructureTopology is highly available but there is a single worker node?

Yes, Ingress will have 2 replicas and they will be randomly scheduled on 2 of the 4 nodes (regardless of being control-plane or true worker nodes)